### PR TITLE
Fixed abi for kotlin top level functions

### PR DESCRIPTION
**Pull request by v-jizhang**
Fix for #2407 
Make buck to generated ABIs for synthetic classes in Kotlin modules.

### CODE REVIEW

**Comment by zpao** [comment]
Let's add a test here :)

File: src/com/facebook/buck/jvm/java/abi/StubJarClassEntry.java [hunk 1]
```diff
@@ -152,11 +152,14 @@ private InputStream openInputStream() {
     return new ByteArrayInputStream(writer.toByteArray());
   }
 
-  private static boolean isAnonymousOrLocalOrSyntheticClass(ClassNode node) {
+  private static boolean isSyntheticClass(ClassNode node) {
     if ((node.access & Opcodes.ACC_SYNTHETIC) == Opcodes.ACC_SYNTHETIC) {
       return true;
     }
+    return false;
```
Comment (zpao) [inline src/com/facebook/buck/jvm/java/abi/StubJarClassEntry.java:159]:
```suggestion
    return (node.access & Opcodes.ACC_SYNTHETIC) == Opcodes.ACC_SYNTHETIC);
```

File: src/com/facebook/buck/jvm/java/abi/StubJarClassEntry.java [hunk 2]
```diff
@@ -106,7 +106,7 @@ public static StubJarClassEntry of(
 
     // The synthetic package-info class is how package annotations are recorded; that one is
     // actually used by the compiler
-    if (!isAnonymousOrLocalOrSyntheticClass(stub)
+    if (!((isSyntheticClass(stub) && !isKotlinModule) || isAnonymousOrLocalClass(stub))
```
Comment (zpao) [inline src/com/facebook/buck/jvm/java/abi/StubJarClassEntry.java:109]: If we can simplify this logic and match formatting, that'd be great.

Comment (v-jizhang) [reply src/com/facebook/buck/jvm/java/abi/StubJarClassEntry.java]: Simplified.

**Pull request merged**